ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java

```diff
@@ -289,26 +283,50 @@ public void notifyTransferMethodAdded(@NonNull final HyperwalletTransferMethod t
     }
 
     @Override
-    public void showTransferMethodFields(@NonNull final List<HyperwalletField> fields) {
+    public void showTransferMethodFields(@NonNull final List<HyperwalletFieldGroup> fields) {
         mDynamicContainer.removeAllViews();
-        int previousView = 0;
+        int previousView;
+
         try {
-            for (final HyperwalletField field : fields) {
-                AbstractWidget widget = WidgetFactory
-                        .newWidget(field, this, getContext(),
-                                mWidgetInputStateHashMap.containsKey(field.getName()) ?
-                                        mWidgetInputStateHashMap.get(field.getName()).getValue() : "",
-                                mCreateTransferMethodButton);
-                if (mWidgetInputStateHashMap.isEmpty() || !mWidgetInputStateHashMap.containsKey(widget.getName())) {
-                    mWidgetInputStateHashMap.put(widget.getName(), widget.getWidgetInputState());
+            Locale locale = new Locale.Builder().setRegion(mCountry).build();
+            // group
+            for (HyperwalletFieldGroup group : fields) {
+                String sectionHeaderText;
+                if (group.getGroupName().equals(HyperwalletFieldGroup.GroupTypes.ACCOUNT_INFORMATION)) {
+                    sectionHeaderText = requireContext().getResources()
+                            .getString(R.string.account_information_section_header, locale.getDisplayName(), mCurrency);
+                } else {
+                    sectionHeaderText = requireContext().getString(requireContext().getResources()
+                            .getIdentifier(group.getGroupName().toLowerCase(Locale.ROOT), "string",
+                                    requireContext().getPackageName()));
                 }
 
-                View widgetView = widget.getView();
-                widgetView.setTag(widget);
-                previousView = placeBelow(widgetView, previousView, true);
-                final String error = mWidgetInputStateHashMap.get(widget.getName()).getErrorMessage();
-                widget.showValidationError(error);
-                mDynamicContainer.addView(widgetView);
+                View sectionHeader = LayoutInflater.from(mDynamicContainer.getContext())
+                        .inflate(R.layout.item_widget_section_header, mDynamicContainer, false);
+                TextView sectionTitle = sectionHeader.findViewById(R.id.section_header_title);
+                sectionTitle.setText(sectionHeaderText);
+                sectionHeader.setId(View.generateViewId());
+                previousView = sectionHeader.getId();
+                mDynamicContainer.addView(sectionHeader);
+
+                // group fields
+                for (final HyperwalletField field : group.getFields()) {
```

Comment:
@peter-joseph  what should we do if there is no fields (they are nulablle) maybe we can add :
if (group.getFields() != null)

Reply:
its a mis configuration